### PR TITLE
Make decoding of `Raw` to configs consistent

### LINKED ISSUE #202: Identically decode from `runtime.RawExtension.Raw` to `api.RsyslogRelpConfig`
**How to categorize this issue?**

/area quality
/kind enhancement

**What would you like to be added**:
 Let's do the decoding from `runtime.RawExtension.Raw` to `api.RsyslogRelpConfig` identically in all places. We can do it through:
-  a helper function. Let's move `decodeRsyslogRelpConfig` to `pkg/apis/rsyslog/helper/helpers.go` 
- use `extensions/pkg/util/serialization.go@Decode`

https://github.com/gardener/gardener-extension-shoot-rsyslog-relp/blob/8d1601a4d5106d8e579de220015780f1d136b4f4/pkg/controller/lifecycle/actuator.go#L60-L63

**Why is this needed**:
Minor code simplification

### PR DESCRIPTION
**Pull request by RadaBDimitrova**
**How to categorize this PR?**

/area quality
/kind enhancement

**What this PR does / why we need it**:
Whenever decoding is needed, it's now done using `runtime.DecodeInto()`.

**Which issue(s) this PR fixes**:
Fixes #202

**Special notes for your reviewer**:
N/A

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by plkokanov** [comment]
https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener-extension-shoot-rsyslog-relp/209/pull-gardener-extension-shoot-rsyslog-relp-e2e-kind/1867148206291292160 passed successfully, however, it seems that the cleanup process was terminated before it could completely finish successfully:

```
Deleting cluster "gardener-local" ...
Deleted nodes: ["gardener-local-control-plane"]
make[1]: Leaving directory '/home/prow/go/src/github.com/gardener/gardener-extension-shoot-rsyslog-relp/gardener'
make: *** [Makefile:149: ci-e2e-kind] Terminated
wrapper.sh] [EARLY EXIT] Exit code was 143, not preserving due to interrupt signal
```

**Review by Kostov6 — COMMENTED** [review]
You can drop `func decodeRsyslogRelpConfig` as it should be no longer in use

**Review by Kostov6 — COMMENTED** [review]
/lgtm

File: pkg/admission/validator/shoot.go [hunk 1]
```diff
@@ -165,7 +169,7 @@ func validateAuditConfigMap(decoder runtime.Decoder, configMap *corev1.ConfigMap
 
 	auditdConfig := &rsyslog.Auditd{}
```
Comment (plkokanov) [inline pkg/admission/validator/shoot.go:171]:
```suggestion
```
Let's also use this PR to remove the unnecessary empty line here

**Review by plkokanov — CHANGES_REQUESTED** [review]
Just one small cosmetic nit, otherwise looks good.

File: pkg/admission/validator/shoot.go [hunk 2]
```diff
@@ -164,8 +168,7 @@ func validateAuditConfigMap(decoder runtime.Decoder, configMap *corev1.ConfigMap
 	}
 
 	auditdConfig := &rsyslog.Auditd{}
-
-	_, _, err := decoder.Decode([]byte(auditdConfigString), nil, auditdConfig)
+	err := runtime.DecodeInto(decoder, []byte(auditdConfigString), auditdConfig)
```
Comment (plkokanov) [inline pkg/admission/validator/shoot.go:171]: Sorry, just noticed this on a second look - could you also use one line for the error checking in all occurrences? 
```suggestion
	if err := runtime.DecodeInto(decoder, []byte(auditdConfigString), auditdConfig); err != nil {
```

**Review by plkokanov — CHANGES_REQUESTED** [review]

**Review by plkokanov — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**